plugins/async-node/core/src/index.ts

```diff
@@ -166,11 +166,16 @@ export class AsyncNodePluginPlugin implements AsyncNodeViewPlugin {
               options.parseNode && result
                 ? options.parseNode(result)
                 : undefined;
-
             if (parsedNode) {
               this.resolvedMapping.set(node.id, parsedNode);
-              view.updateAsync();
+            } else {
+              // If parsedNode is undefined, update the resolvedMapping with the existing node (supporting null | undefined) as of now
+              // This is to ensure that the node is not resolved again
+              if (!this.resolvedMapping.has(node.id)) {
```

Comment:
This plugin, if registered, will live as long as the player instance it's registered to is still active.
That means as we repeated update UI, this check will only be true once.

I'm not 100% sure your original solution _won't_ work, i'd write some tests to make sure the solution is okay through multiple updates

Reply:
Changed the business logic to such the node is resolved irrespective of the type which ensures that the consumer can expect the node to be resolved irrespective of the input they provide us with. Also updated respective test case such that we ensure nothing is resolved and also that the resolution happens only once and after the promise has been resolved.

Reply:
I think we should check that returning the same async node doesn't impact future updates, I think you could do so by manually calling the `update` function in the resolver, see what happens if you try to update again after the resolve with `null`/`undefined` has been processed

Reply:
I think you're not looking at the updated code .. This is the updated code 
![image](https://github.com/player-ui/player/assets/171299777/59e9d6ac-c149-4aa1-a4e0-4bb1c771f536)